nodes/analyzer/component_data.py

```diff
@@ -0,0 +1,309 @@
+        default="Edge",
+        update=update_mode)
+    center_mode: EnumProperty(
+        name="Center",
+        items=pols_origin_modes[:3],
+        default="Median Center",
+        update=update_mode)
+    pols_origin_mode: EnumProperty(
+        name="Origin",
+        items=pols_origin_modes,
+        default="Median Center",
+        update=update_mode)
+    matrix_track: EnumProperty(
+        name="Normal",
+        items=matrix_track_modes,
+        default="Z",
+        update=update_mode)
+    matrix_normal_up: EnumProperty(
+        name="Up",
+        items=matrix_track_modes,
+        default="Y",
+        update=update_mode)
+    matrix_normal: EnumProperty(
+        name="Edge",
+        items=matrix_normal_modes,
+        default="X",
+        update=update_mode)
+
+    def actual_mode(self):
+        if self.mode == 'Verts':
+            component_mode = self.vertex_mode
+        elif self.mode == 'Edges':
+            component_mode = self.edge_mode
+        else:
+            component_mode = self.face_mode
+        return component_mode
+
+    def draw_label(self):
+        text = "CA: " + self.mode + " "+ self.actual_mode()
+        return text
+
+    def draw_buttons(self, context, layout):
+        layout.prop(self, "mode", expand=True)
+
+        if self.mode == 'Verts':
+            layout.prop(self, "vertex_mode", text="")
+        elif self.mode == 'Edges':
+            layout.prop(self, "edge_mode", text="")
+        elif self.mode == 'Faces':
+            layout.prop(self, "face_mode", text="")
+        info = modes_dicts[self.mode][self.actual_mode()]
+        local_ops = info[3]
+        out_ops = info[2]
+        op_dict = {
+        'c': 'center_mode',
+        's': 'sum_items',
+        'o': 'origin_mode',
+        'q': 'pols_origin_mode',
+        'm': 'matrix_track',
+        'u': 'matrix_normal_up',
+        'n': 'matrix_normal',
+        't': 'tangent_mode',
+        }
+        for op in local_ops:
+            if op in op_dict:
+                layout.prop(self, op_dict[op])
+        if not 'u' in out_ops:
+            layout.prop(self, 'split')
+        else:
+            layout.prop(self, 'wrap')
+
+
+    def sv_init(self, context):
+        inew = self.inputs.new
+        inew('SvVerticesSocket', "Vertices")
+        inew('SvStringsSocket', "Edges")
+        inew('SvStringsSocket', "Faces")
+
+        onew = self.outputs.new
+        onew('SvStringsSocket', "Vals")
+        onew('SvVerticesSocket', "Faces")
+        onew('SvVerticesSocket', "Mask")
+
+        self.update_mode(context)
+
+    def process(self):
+        if not any(output.is_linked for output in self.outputs):
+            return
+        modes_dict = modes_dicts[self.mode]
+        component_mode = self.actual_mode()
+        output_ops = modes_dict[component_mode][2]
+        local_ops = modes_dict[component_mode][3]
+        func_inputs = modes_dict[component_mode][4]
+        func = modes_dict[component_mode][5]
```

Comment:
Just alternative way:
```Python
_, _, output_ops, local_ops, func_inputs, func, *_ = modes_dict[component_mode]
```

Reply:
Maybe use a normal class instead of tuple?... Tuples with more than 3-4 parameters tend to be too fragile - it's too easy to confuse the order of parameters.
Or named tuples, as an option...

Reply:
Looks like `namedtuple` is also deprecated: https://docs.python.org/3.7/library/collections.html#collections

> Deprecated since version 3.3, will be removed in version 3.9: Moved Collections Abstract Base Classes to the collections.abc module. For backwards compatibility, they continue to be visible in this module through Python 3.8.

Reply:
Probably new way is using `dataclasses` module.
https://docs.python.org/3.7/library/dataclasses.html

Reply:
@Durman I tend to understand that phrase
> Deprecated since version 3.3, will be removed in version 3.9: Moved Collections Abstract Base Classes to the collections.abc module.

as "Collections ABC are deprecated in collections module, and will be moved to collections.abc; but other classes and methods from collections module are okay".

Reply:
I would need to create the classes from the original table-looking dicts so it is another step for a dict that is only called 1 time per update...